spec/features/user_can_signup_spec.rb

```diff
@@ -1,80 +1,107 @@
-  context "Form data" do
+RSpec.feature 'Signup', type: :feature do
+  context 'Form data' do
 
-    scenario "it has an email field" do
+    scenario 'it has an email field' do
       go_to_homepage
 
       expect(page).to have_field("user[email]") 
     end
 
-    scenario "it has a password field" do
+    scenario 'it has a password field' do
       go_to_homepage
 
       expect(page).to have_field("user[password]") 
     end
+
+    scenario 'it has a name field' do
+      go_to_homepage
+
+      expect(page).to have_field("user[name]") 
+    end
+
+    scenario 'it has a password confirmation field' do
+      go_to_homepage
+
+      expect(page).to have_field('user[password_confirmation]') 
+    end
   end
 
-  context "User must fill in a valid password" do
-    scenario "user left password field empty" do
-      sign_up(email: "test@test.com", password: "")
+  context 'User must fill in a valid password' do
+    scenario 'user left password field empty' do
+      sign_up(name: 'test', email: "test@test.com", password: '', password_confirmation: '')
 
-      expect(page).to have_content("Password can't be blank")
-      expect(page).to have_content("Password is too short (minimum is 6 characters)")
+      expect(page).to have_content('Password can\'t be blank')
+      expect(page).to have_content('Password is too short (minimum is 6 characters)')
     end
 
     scenario "user provides a password sorter than 6 characters" do
-      sign_up(email: "test@test.com", password: "12345")
+      sign_up(name: 'test', email: "test@test.com", password: "12345", password_confirmation: '12345')
 
       expect(page).to have_content("Password is too short (minimum is 6 characters)")
     end
 
     scenario "user provides a password longer than 10 characters" do
-      sign_up(email: "test@test.com", password: "12345678910")
+      sign_up(name: 'test', email: "test@test.com", password: "12345678910", password_confirmation: '12345678910')
 
       expect(page).to have_content("Password is too long (maximum is 10 characters)")
     end
+
+    scenario "user must provide passwords that match" do
+      sign_up(name: 'test', email: 'test@test.com', password: '', password_confirmation: '1234567')
+
+      expect(page).to have_content('Password confirmation doesn\'t match Password')
+    end
   end
 
   context "User must provide a valid email address" do
     scenario "user left email field empty" do 
-      sign_up(email: "", password: "123456789")
+      sign_up(name: 'test', email: '', password: "123456789", password_confirmation: '123456789')
 
       expect(page).to have_content("Email can't be blank")
     end
 
     scenario "user provides an email address that has already been taken" do
-      sign_up(email: "test@test.com", password: "123456789")
+      sign_up(name: 'test', email: "test@test.com", password: "123456789", password_confirmation: '123456789')
 
       expect(page).to have_current_path("/posts")
```

Comment:
No need for line 68 - tested already on line 106

Reply:
We do need the line, as we are testing for duplicate email addresses at signup.